packages/botonic-api/src/data-provider/local-data-provider.ts

```diff
@@ -22,143 +22,117 @@ export class LocalDevDataProvider implements DataProvider {
+    const to = limit !== 0 ? from + limit : undefined
+    return users.slice(from, to)
   }
 
   getUser(userId: string): User | undefined {
     const path = this.createPath([this.paths.USERS, userId])
-    try {
-      this.db.reload()
-      return this.db.exists(path) ? this.db.getObject<User>(path) : undefined
-    } catch (e) {
-      console.error(`Error fetching user with ID '${userId}' from local DB`, e)
-    }
-    return undefined
+    this.db.reload()
+    return this.db.exists(path) ? this.db.getObject<User>(path) : undefined
   }
 
   saveUser(user: User): User {
     const path = this.createPath([this.paths.USERS, user.id])
-    try {
-      this.db.reload()
-      this.db.push(path, user, false)
-    } catch (e) {
-      console.error(`Error saving user with ID '${user.id}' to local DB`, e)
+    this.db.reload()
+    if (!this.db.exists(path)) {
+      this.db.push(path, user)
     }
     return user
   }
 
   updateUser(user: User): User {
     const path = this.createPath([this.paths.USERS, user.id])
-    try {
-      this.db.reload()
+    this.db.reload()
+    if (this.db.exists(path)) {
       this.db.push(path, user, true)
-    } catch (e) {
-      console.error(`Error updating user with ID '${user.id}' to local DB`, e)
     }
+
     return user
   }
 
   getEvents(limit = 10, offset = 0): BotonicEvent[] {
     const path = this.paths.EVENTS
-    try {
-      this.db.reload()
-      const eventList = this.db.getObject<Record<string, BotonicEvent>>(path)
-      const events = Object.values(eventList)
-      const from = offset * limit
-      const to = limit !== 0 ? from + limit : undefined
-      return events.slice(from, to)
-    } catch (e) {
-      console.error('Error fetching events from local DB', e)
-    }
-    return []
+    this.db.reload()
+    const eventList = this.db.getObject<Record<string, BotonicEvent>>(path)
+    const events = Object.values(eventList)
+    const from = offset
+    const to = limit !== 0 ? from + limit : undefined
+    return events.slice(from, to)
   }
 
   getEvent(id: string): BotonicEvent | undefined {
     const path = this.createPath([this.paths.EVENTS, id])
-    try {
-      this.db.reload()
-      return this.db.exists(path)
-        ? this.db.getObject<BotonicEvent>(path)
-        : undefined
-    } catch (e) {
-      console.error(`Error fetching event with ID '${id}' from local DB`, e)
-    }
-    return undefined
+    this.db.reload()
+    return this.db.exists(path)
+      ? this.db.getObject<BotonicEvent>(path)
+      : undefined
   }
 
   saveEvent(event: BotonicEvent): BotonicEvent {
     const path = this.createPath([this.paths.EVENTS, event.eventId])
-    try {
-      this.db.reload()
-      this.db.push(path, event, false)
-    } catch (e) {
-      console.error(
-        `Error saving event with user ID '${event.userId}' to local DB`,
-        e
-      )
+    this.db.reload()
+    if (!this.db.exists(path)) {
+      this.db.push(path, event)
+    }
+    return event
```

Comment:
Probably not for this PR, but not sure whether we could have commong `update`/`delete`/ ... methods that follow this common pattern. This way, you only need to pass the path and the event to a common method to do the rest.

Reply:
I think it won't be needed in this situation since all methods differ a little bit from each other and the common method would be too complicated adding specific behaviour for each event.